lib/linux.js

```diff
@@ -41,18 +41,23 @@ var LinuxBuilder = {
       linux.archName = 'x86_64';
     }
 
-    var tmpFolder = path.join( os.tmpDir(), linux.executable );
+    var outFilename = linux.executable + '-' +
+                      linux.version + '-' + linux.archName + '.' +
+                      linux.target;
+    var tmpFolder = path.join( os.tmpDir(), outFilename );
     _createFolders( tmpFolder );
 
     _createDesktopEntry( tmpFolder, linux );
 
     var scripts = _createScripts( linux );
 
-    _copyApp( options, tmpFolder );
+    fs.copySync( options.appPath,
+      path.join( tmpFolder, 'opt', options.config.linux.executable ) );
 
-    _buildPackage( options, scripts, tmpFolder );
+    var destination = path.join( options.out, outFilename );
+    _buildPackage( options, scripts, tmpFolder, destination );
 
-    callback();
+    callback( null, destination );
```

Comment:
Callback returns out path (as electron-packager does). It is used in electron-complete-builder to publish artifact.